### ISSUE TITLE
Enclose YAML value in quotes (if no negative consequences) to enable downstream parsing via `yaml.safe_load()`

### ISSUE THREAD

**Issue by eecavanna** [microbiomedata/nmdc-runtime#730]
I am working on a centralized documentation website that will pull raw documentation files from multiple NMDC repositories, build a single website from them, and deploy it to GitHub Pages.

While trying to pull the `mkdocs.yml` file from this repository, I ran into a 2-layer issue. This ticket is about the second layer.

The first layer was that the website URL is hard-coded in the YAML file (in the `site_url` field). I am not surprised by that (I think I'd do it that way, myself), but it is interfering with what I'm trying to do for the centralized documentation website. So, my plan was to read the YAML file into a Python script, modify the values I want to modify, then dump the result out to a file, which would then be fed to mkdocs.

That brings me to layer two. While trying to read that file using PyYAML in various ways, PyYAML complained about one line of the YAML file:

https://github.com/microbiomedata/nmdc-runtime/blame/38b3c7375d95206af0e7b6877f547429020fcc95/mkdocs.yml#L59

Here is how PyYAML behaved in four scenarios:

### YAML as is, reading with `safe_load`

```yaml
          format: !!python/name:pymdownx.superfences.fence_code_format
```

```py
mkdocs_config = yaml.safe_load(f)
```

```py
yaml.constructor.ConstructorError: could not determine a constructor for the tag 'tag:yaml.org,2002:python/name:pymdownx.superfences.fence_code_format'
  in "test/data/mkdocs.yml", line 59, column 19
```

### YAML as is, reading with `unsafe_load`

```yaml
          format: !!python/name:pymdownx.superfences.fence_code_format
```

```py
mkdocs_config = yaml.unsafe_load(f)
```

```py
yaml.constructor.ConstructorError: while constructing a Python object
cannot find module 'pymdownx.superfences' (No module named 'pymdownx')
  in "test/data/mkdocs.yml", line 59, column 19
```

### YAML with surrounding double quotes added, reading with `safe_load`

```yaml
          format: "!!python/name:pymdownx.superfences.fence_code_format"
```

```py
mkdocs_config = yaml.safe_load(f)
```

```py
{'site_name': 'nmdc-runtime docs', 'site_description': 'Documentation for the National Microbiome Data Collaborative (NMDC) Runtime System and API', 'site_url': 'https://microbiomedata.github.io/nmdc-runtime/', 'repo_url': 'https://github.com/microbiomedata/nmdc-runtime/', 'edit_uri_template': 'blob/main/docs/{path}', 'extra_css': ['stylesheets/extra.css'], 'nav': ['index.md', {'Tutorials': ['tutorials/auth.md', 'tutorials/json.md', 'tutorials/metadata-in.md']}, {'How-to guides': ['howto-guides/author-changesheets.md', 'howto-guides/create-triggers.md', 'howto-guides/claim-and-run-jobs.md', 'howto-guides/update-sensors-ops.md', 'howto-guides/release-process.md', 'howto-guides/improving-search-api.md', 'nb/get_data.ipynb', 'nb/queue_and_trigger_data_jobs.ipynb', 'nb/wf_automation.ipynb', {'Specific Jobs': ['howto-guides/jobs/gold-translation-etl.md']}]}, {'Reference': ['contributing-docs.md', 'admin.md']}, {'Explanation': ['explanation/journeys.md', 'explanation/identifiers.md']}], 'theme': {'name': 'material', 'custom_dir': 'mkdocs_overrides', 'palette': [{'media': '(prefers-color-scheme: light)', 'scheme': 'default', 'toggle': {'icon': 'material/toggle-switch-off-outline', 'name': 'Switch to dark mode'}}, {'media': '(prefers-color-scheme: dark)', 'scheme': 'slate', 'toggle': {'icon': 'material/toggle-switch', 'name': 'Switch to light mode'}}]}, 'markdown_extensions': ['admonition', 'attr_list', 'footnotes', 'md_in_html', {'toc': {'permalink': True}}, {'pymdownx.superfences': {'custom_fences': [{'name': 'mermaid', 'class': 'mermaid', 'format': '!!python/name:pymdownx.superfences.fence_code_format'}]}}], 'plugins': ['search', {'mermaid2': {'arguments': {'securityLevel': 'loose'}}}, 'mkdocs-jupyter']}
```
> No problem.

### YAML with surrounding double quotes added, reading with `unsafe_load`

```yaml
          format: "!!python/name:pymdownx.superfences.fence_code_format"
```

```py
mkdocs_config = yaml.unsafe_load(f)
```

```py
{'site_name': 'nmdc-runtime docs', 'site_description': 'Documentation for the National Microbiome Data Collaborative (NMDC) Runtime System and API', 'site_url': 'https://microbiomedata.github.io/nmdc-runtime/', 'repo_url': 'https://github.com/microbiomedata/nmdc-runtime/', 'edit_uri_template': 'blob/main/docs/{path}', 'extra_css': ['stylesheets/extra.css'], 'nav': ['index.md', {'Tutorials': ['tutorials/auth.md', 'tutorials/json.md', 'tutorials/metadata-in.md']}, {'How-to guides': ['howto-guides/author-changesheets.md', 'howto-guides/create-triggers.md', 'howto-guides/claim-and-run-jobs.md', 'howto-guides/update-sensors-ops.md', 'howto-guides/release-process.md', 'howto-guides/improving-search-api.md', 'nb/get_data.ipynb', 'nb/queue_and_trigger_data_jobs.ipynb', 'nb/wf_automation.ipynb', {'Specific Jobs': ['howto-guides/jobs/gold-translation-etl.md']}]}, {'Reference': ['contributing-docs.md', 'admin.md']}, {'Explanation': ['explanation/journeys.md', 'explanation/identifiers.md']}], 'theme': {'name': 'material', 'custom_dir': 'mkdocs_overrides', 'palette': [{'media': '(prefers-color-scheme: light)', 'scheme': 'default', 'toggle': {'icon': 'material/toggle-switch-off-outline', 'name': 'Switch to dark mode'}}, {'media': '(prefers-color-scheme: dark)', 'scheme': 'slate', 'toggle': {'icon': 'material/toggle-switch', 'name': 'Switch to light mode'}}]}, 'markdown_extensions': ['admonition', 'attr_list', 'footnotes', 'md_in_html', {'toc': {'permalink': True}}, {'pymdownx.superfences': {'custom_fences': [{'name': 'mermaid', 'class': 'mermaid', 'format': '!!python/name:pymdownx.superfences.fence_code_format'}]}}], 'plugins': ['search', {'mermaid2': {'arguments': {'securityLevel': 'loose'}}}, 'mkdocs-jupyter']}
```
> No problem.

# Task

- [ ] Determine whether adding quotes around the `format` value has an adverse effects on the behavior of MkDocs
- [ ] If it does not, add the double quotes to the file so it can be parsed using `safe_load`

**Comment by eecavanna** [comment]
I assigned it to @dwinston because Git shows him as the only person that has edited the file so far. CC: @PeopleMakeCulture

**Comment by ssarrafan** [comment]
@dwinston @eecavanna will this be actively worked on in the next 2 weeks?

**Comment by eecavanna** [comment]
I, personally, don't plan to work on it during the next couple weeks.

**Comment by ssarrafan** [comment]
No updates. Removing from sprint, adding unfinished/backlog labels. FYI @eecavanna @dwinston

**Comment by eecavanna** [comment]
When using `yaml.unsafe_load()`, the error message says that something in a module named `pymdownx.superfences` cannot be found.

I found the code for that module, here: https://github.com/facelessuser/pymdown-extensions/blob/main/pymdownx/superfences.py (docs: https://facelessuser.github.io/pymdown-extensions/extensions/superfences/)

I don't see that package listed in any of the files in this repo's `requirements/` directory. 🤷 In other words, I don't know how pip would know to install it.

**Comment by eecavanna** [comment]
Turns out the package that gets installed by pip is not called `pymdownx`, but, rather, `pymdown-extensions`.

I do see the latter in `requirements/main.txt`, here: https://github.com/microbiomedata/nmdc-runtime/blob/1e9f9c28ab849e9cce9996c3559ff585bb32a4f7/requirements/main.txt#L558

I will install that package downstream (maybe via some non-transitive dependency I neglected to install—TBD). Closing.
